cpu/nrf52/cpu.c

```diff
@@ -68,14 +68,13 @@ void cpu_init(void)
 #ifdef SOFTDEVICE_PRESENT
     softdevice_handler_init(NRF_CLOCK_LFCLKSRC_XTAL_20_PPM, &_ble_evt_buffer,
             BLE_STACK_EVT_MSG_BUF_SIZE, NULL);
-#endif
-    /* call cortexm default initialization */
-    cortexm_init();
 
-#ifdef SOFTDEVICE_PRESENT
     /* fixup swi0 (used as softdevice PendSV trampoline) */
     NVIC_EnableIRQ(SWI0_EGU0_IRQn);
     NVIC_SetPriority(SWI0_EGU0_IRQn, 6);
+#else
+    /* call cortexm default initialization */
+    cortexm_init();
```

Comment:
Hmmm... are you sure all the functions there are unnecessary? You'll also need to increase the priority as you did before isn't it? Or with this configuration it works normally?

Reply:
Well It works so far but I guess I could still add some part of the cortexm_init.
But since we have no clue of what the softdevice is doing maybe we should stay like this...
What do you think ?

Reply:
I'd say as long as it works properly (and several initialisations are certainly done by the softdevice), we can leave it as is. I'll try to make some extensive tests soon.

Reply:
Great ! many thanks @kYc0o